### PR TITLE
Add support for custom fields in the log messages for unit test logger

### PR DESCRIPTION
**Pull request by shobhit85**
This change dumps logger fields out added to a logger for unit tests. Fields are supported for other default outputs like, `stdout` `stderr` etc. but unit tests use `logbuffer` output which did not support logger fields.

@zquestz @jpoler @wallyqs

### CODE REVIEW

**Comment by wallyqs** [comment]
LGTM 👍 let's do this

File: unittest/unittest.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ const (
 	// This format string is used to replace the time stamp with a fixed number
 	// representing the number of log lines contained in the buffer. We fake
 	// this a bit by using epoch and mutating the real time stamp.
-	formatStr = "%color:class%[%hour%:%minute%:%second%.%nanosecond% " +
+	defaultFormatStr = "%color:class%[%hour%:%minute%:%second%.%nanosecond% " +
```
Comment (wallyqs) [inline unittest/unittest.go]: IMO time at the beginning looks better for sorting with basic unix tools

Comment (shobhit85) [inline unittest/unittest.go]: Agree. May be change that in separate PR?

Comment (wallyqs) [inline unittest/unittest.go]: 👍
